worker/worker-execution/score-worker-execution-impl/src/main/java/io/cloudslang/worker/execution/services/ExecutionServiceImpl.java

```diff
@@ -547,22 +547,48 @@ private static boolean useDefaultGroup(Execution execution) {
         return useDefaultGroup;
     }
 
-    public static void postExecutionSettings(Execution execution) {
-        // Decide on Group
+    public void postExecutionSettings(Execution execution) {
+        setWorkerGroup(execution);
+
+        //if there is a request to change the running execution plan id, we update the execution to the new execution plan ID
+        Long requestForChangingExecutionPlan = execution.getSystemContext().pullRequestForChangingExecutionPlan();
+        if (requestForChangingExecutionPlan != null) {
+            execution.setRunningExecutionPlanId(requestForChangingExecutionPlan);
+        }
+    }
+
+    private void setWorkerGroup(Execution execution) {
+        //get group from system context
         String group = (String) execution.getSystemContext().get(TempConstants.ACTUALLY_OPERATION_GROUP);
```

Comment:
Static imports

Reply:
since in this class there are multiple constants used from different other classes I believe that this would break readability. It's useful to know which constants are from TempConstants and which from EventConstants for example

Reply:
How is that useful? Besides having more code with same functionality.